scripts/ci/libraries/_build_images.sh

```diff
@@ -559,12 +558,27 @@ function build_ci_image() {
         echo >&2
         exit 1
     fi
+    if [[ -n ${SPIN_PID:=""} ]]; then
+        kill -HUP "${SPIN_PID}" || true
+        wait "${SPIN_PID}" || true
+        echo > "${DETECTED_TERMINAL}"
```

Comment:
What values can the variable ``DETECTED_TERMINAL`` have?  I can see that this variable is always set to `/dev/tty`. Did I miss something?

Reply:
it can be /dev/tty if we cannot detect therminal with $(tty) (which happens inside pre-commit on git commit (it is always non-interactive by design)

Reply:
it can be /dev/tty if we cannot detect therminal with $(tty) (which happens inside pre-commit on git commit (it is always non-interactive by design)
```
elif [[ ${DETECTED_TERMINAL:=$(tty)} != "not a tty" ]]; then
````

Reply:
The whole thing is about trying to print progress to most likely terminal i fbuild happens insde pre-commit inside `git commit command - it won't always succeed but in vast majority of cases of developer workstation  - it will